src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -226,17 +221,44 @@ class JooqBurdenEstimateRepository(
         val responsibilityInfo = getResponsibilityInfo(modellingGroup.id, touchstoneId, scenarioId)
 
         val set = getBurdenEstimateSet(setId)
-        if (set.status != BurdenEstimateSetStatus.EMPTY)
+
+        if (set.status == BurdenEstimateSetStatus.COMPLETE)
         {
-            throw OperationNotAllowedError("This burden estimate set already contains estimates." +
+            throw OperationNotAllowedError("This burden estimate set is marked as complete." +
                     " You must create a new set if you want to upload any new estimates.")
         }
 
-        BurdenEstimateWriter(dsl, setId).addEstimatesToSet(estimates, responsibilityInfo.disease)
+        if (set.type.type == BurdenEstimateSetTypeCode.STOCHASTIC)
+        {
+            populateStochasticBurdenEstimateSet(set, responsibilityInfo.disease, estimates)
+        }
+        else
+        {
+            populateCentralBurdenEstimateSet(set, responsibilityInfo.disease, estimates)
+        }
+
         updateCurrentBurdenEstimateSet(responsibilityInfo.id, setId, set.type.type)
+    }
+
+    private fun populateCentralBurdenEstimateSet(set: BurdenEstimateSet, disease: String, estimates: Sequence<BurdenEstimateWithRunId>)
+    {
+        CentralBurdenEstimateWriter(dsl, set.id).addEstimatesToSet(estimates, disease)
         dsl.update(Tables.BURDEN_ESTIMATE_SET)
                 .set(Tables.BURDEN_ESTIMATE_SET.STATUS, "complete")
-                .where(Tables.BURDEN_ESTIMATE_SET.ID.eq(setId))
+                .where(Tables.BURDEN_ESTIMATE_SET.ID.eq(set.id))
+                .execute()
+    }
+
+    private fun populateStochasticBurdenEstimateSet(set: BurdenEstimateSet, disease: String,
+                                                    estimates: Sequence<BurdenEstimateWithRunId>)
+    {
+        StochasticBurdenEstimateWriter(AnnexJooqContext().dsl,
+                set.id, BurdenEstimateWriter(dsl, set.id))
+                .addEstimatesToSet(estimates, disease)
+
+        dsl.update(Tables.BURDEN_ESTIMATE_SET)
+                .set(Tables.BURDEN_ESTIMATE_SET.STATUS, "partial")
```

Comment:
Setting the status to partial here rather than complete was based on the assumption that we might want to make multiple calls to populate a stochastic set from multiple files. I'm not sure if this assumption is correct?

Reply:
I think it makes more sense to address partial uploads for both central and stochastic estimates in one go, when we take a look at chunked uploads (c.f. https://vimc.myjetbrains.com/youtrack/issue/VIMC-1248). For now, I think it would be better to leave the behaviour the same for both central and stochastic uploads.